tests/yast2_gui/yast2_firewall_set_default_zone.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: YaST2 Firewall UI test checks verious configurations and settings of firewall
+# Make sure yast2 firewall can opened properly. Configurations can be changed and written correctly.
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use network_utils 'iface';
+use YaST::Module;
+use version_utils "is_sle";
+
+sub verify_interface {
+    my (%args) = @_;
+    validate_script_output("firewall-cmd --list-interfaces --zone=$args{zone}", sub { m/$args{device}/ }, proceed_on_failure => 1);
+}
+
+sub verify_default_zone {
+    my (%args) = @_;
+    validate_script_output("firewall-cmd --get-default-zone", sub { m/$args{zone}/ }, proceed_on_failure => 1);
+}
+
+sub change_interface_zone {
+    my (%args) = @_;
+    $testapi::distri->get_firewall()->select_interfaces_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->set_interface_zone($args{device}, $args{zone});
+    save_screenshot;
+}
+
+sub set_default_zone {
+    my $zone = shift;
+    $testapi::distri->get_firewall()->select_zones_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->set_default_zone($zone);
+    save_screenshot;
+}
+
+sub run {
+    my $self = shift;
+    my $iface = iface();
+
+    $self->select_serial_terminal();
+    verify_default_zone(zone => "public");
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    $testapi::distri->get_firewall()->select_interfaces_page();
+    save_screenshot;
+    # TW set default zone with iface together but sle not, so we need update iface's zone to trusted zone firstly
+    if (is_sle()) {
```

Comment:
we don't use conditions in the test layer, let's create a new test module for tumbleweed that we schedule before this one to change the interface, so later when we move the default, all is fine.

Reply:
I do not think make extra module is good idea since this part of code is combine with current specific test check logic(will set trusted zone to default and check the result) .

Reply:
then you need to create two different modules checking two different results, so the thing with libyui-rest-api is that we don't use `if/else` as they have snow ball effect, and other contributors keep adding conditions and the code will end unreadable. For that seems to me more simple to add this extra part as a new test module and then validation are the same, at the end in the test you want to have interface and zone aligned.